### PR TITLE
Updated to version 3.1.1

### PR DESCRIPTION
**Pull request by acarapetis**
Not sure why the automatic robot didn't do this last year, but it'd be nice to have an up-to-date version on conda-forge; so I did it manually. Includes changes to dependencies (both runtime and the new build system.)

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by acarapetis** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-{% set name = "Flask-SQLAlchemy" %}
-{% set version = "3.0.3" %}
+{% set name = "flask_sqlalchemy" %}
```
Comment (synapticarbors) [inline recipe/meta.yaml]: Please do not change the name

File: recipe/meta.yaml [hunk 2]
```diff
@@ -17,13 +17,13 @@ build:
 
 requirements:
   host:
-    - python >=3.7
+    - python >=3.8
     - pip
-    - pdm-pep517 >=1.0.0
+    - flit-core =3.*
```
Comment (synapticarbors) [inline recipe/meta.yaml]: `- flit-core <4` to match the upstream `pyproject.toml`

**Review by synapticarbors — CHANGES_REQUESTED** [review]

**Comment by synapticarbors** [comment]
@acarapetis -- thanks for catching that we don't have the latest version. I'm not sure why the bots didn't automatically submit a PR. I had a few changes, but otherwise looks good.

File: recipe/meta.yaml (hunk 1 above)
Comment (acarapetis) [reply recipe/meta.yaml]: The PyPI name (or at least download URL) seems to have changed - build was failing before I did this. I guess I should hardcode the `flask_sqlalchemy` name in `source.url` but leave it as `Flask-SQLAlchemy` here?

**Review by synapticarbors — APPROVED** [review]

**Pull request merged**